### PR TITLE
Split config-host-main into remove, list, add commands

### PR DESCRIPTION
**Pull request by harshavardhana**
This PR also adds convenient aliases for

 - remove as `rm`
 - list as `ls`
 - add as `a`

Also brings much needed cleanup.

### CODE REVIEW

File: cmd/config-host-add.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+USAGE:
+  {{.HelpName}} ALIAS URL ACCESS-KEY SECRET-KEY [API]
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Add Amazon S3 storage service under "mys3" alias. For security reasons turn off bash history momentarily.
+     $ set +o history
+     $ {{.HelpName}} add mys3 https://s3.amazonaws.com \
```
Comment (vadmeste) [inline cmd/config-host-add.go]: 'add' is not needed here (it is already shown with {{.HelpName}})

File: cmd/config-host-add.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+                 BKIKJAA5BMMU2RHO6IBB V8f1CwQqAcwo80UEIJEjc5gVQUSSx5ohQ9GSrr12
+     $ set -o history
+
+  2. Add Amazon S3 accelerated storage service under "mys3-accel" alias. For security reasons turn off bash history momentarily.
+     $ set +o history
+     $ {{.HelpName}} add mys3-accel https://s3-accelerate.amazonaws.com \
```
Comment (vadmeste) [inline cmd/config-host-add.go]: Same as above.

File: cmd/config-host-list.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+	var hosts []hostMessage
+	for k := range conf.Hosts {
+		if len(k) > maxAlias {
+			maxAlias = len(k)
+		}
+	}
+
+	for k, v := range conf.Hosts {
+		// If specific alias is requested, only print the requested one.
+		if alias != "" && alias == k {
```
Comment (vadmeste) [inline cmd/config-host-list.go]: I think it is more readable if we move out this block outside of for, besides it is better to print an error message when alias is specified but not found

```
if alias != "" {
  if v, ok := conf.Hosts[alias] && ok {
    hosts = []hostsMessage{{  .... }}
  } else {
     error("no alias is found")
  }
}
```

File: cmd/config-host-remove.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+import (
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/console"
+)
+
+var configHostRemoveCmd = cli.Command{
+	Name:            "remove",
+	ShortName:       "rm",
+	Usage:           "Remove hosts from configuration file.",
```
Comment (vadmeste) [inline cmd/config-host-remove.go]: You mean "remove a host" ?

File: cmd/config-host-add.go (hunk 1 above)
Comment (harshavardhana) [inline cmd/config-host-add.go]: ah yes..

File: cmd/config-host-list.go (hunk 3 above)
Comment (harshavardhana) [inline cmd/config-host-list.go]: We already skip that part in checkSyntaxt() where we throw an error when alias is wrong.

File: cmd/config-host-remove.go (hunk 4 above)
Comment (harshavardhana) [inline cmd/config-host-remove.go]: Yes 'a host'

**Pull request merged**